jcl/src/java.base/share/classes/openj9/internal/foreign/abi/InternalDowncallHandler.java

```diff
@@ -123,7 +125,21 @@ void clear() {
 		}
 	}
 
-	private final ThreadLocal<HeapArgInfo> heapArgInfo;
+	/* heapArgInfo holds a per-thread stack of HeapArgInfo structures.
+	 * Each entry tracks Java references to heap-allocated memory segments
+	 *
+	 * createHeapArgInfo controls when a new entry should be created.
```

Comment:
I'm looking for this comment to explain more of the why and less of the what. As per the coding standards:
> Comments are used to explain the intent of the code, not how it works. Well-written code shouldn't require comments to explain how it works. Comments should explain why it is necessary.

So for heapArgInfo you can just explain that its needed for recursive downcalls and for createHeapArgInfo that it is needed since there may be multiple pointer arguments for a downcall.